include/oneapi/dpl/internal/distributed_ranges_impl/sp/algorithms/sort.hpp

```diff
@@ -112,7 +112,7 @@ sort(R&& r, Compare comp = Compare())
 
     T* medians = sycl::malloc_device<T>(n_segments * n_splitters, sp::devices()[0], sp::context());
 
-    for (auto&& [segment_id_, segment] : stdrng::views::enumerate(segments))
+    for (auto&& [segment_id_, segment] : dr::__detail::make_enumerate(segments))
```

Comment:
It turns out that we are using a functional approach for the sake of a functional approach...
I would suggest a little bit re-writing this loop is following way:

```
for (auto segment_id = 0, n =  stdrng::size(segment); segment_id < n; ++segment_id)
{
         auto segment = segments[segment_id ];
.......
```

Reply:
@MikeDvorskiy , thanks for looking into this. Looks acceptable, although it is more lines of code. 

Are other places with enumarate also easily changeable? Do you think it is more clean than having `make_enumerate' or does it fixes some problems which `make_enumerate` has (if it has)?

Enumerate is quite common programming pattern making code shorter and more readable.

I don't object of changing all places where there is enumerate to some code like this if it is needed to add C++20 support. Please feel free to clone my change to your branch and continue adding any changes you think may be good for C++20 support.